code/modules/antagonists/traitor/datum_traitor.dm

```diff
@@ -1,5 +1,19 @@
-#define TRAITOR_HUMAN "human"
-#define TRAITOR_AI   "AI"
+/// Chance that the traitor could roll hijack if the pop limit is met.
+#define HIJACK_PROB 10
+/// Hijack is unavailable as a random objective below this player count.
+#define HIJACK_MIN_PLAYERS 30
+
+/// Chance the traitor gets a martyr objective instead of having to escape alive, as long as all the objectives are martyr compatible.
+#define MARTYR_PROB 20
+
+/// Chance the traitor gets a kill objective. If this prob fails, they will get a steal objective instead.
+#define KILL_PROB 50
+/// If a kill objective is rolled, chance that it is to destroy the AI.
+#define DESTROY_AI_PROB (100 / GLOB.joined_player_list.len)
```

Comment:
Make this `DESTROY_AI_PROB()`, I don't like when stuff like this lies that its a constant.

Reply:
also, what happens if GLOB.joined_player_list.len is 0 somehow?

Reply:
It is merely a preprocessor macro. It performs no logic, and instead performs a cut-and-paste text replacement on compile. If joined_player_list.len is 0, this could would never have been run as there would be no player to add this antag datum to. If we're adding antag datums to players when joined_player_list is empty, we have bigger issues.

Reply:
Updated macro